### DISCUSSION TITLE
Compilation failure under CentOS 7.9 using Cmake 3.14.5

### DISCUSSION THREAD

**Discussion by hadesya** [Chia-Network/bladebit#106] [Q&A] [answered]
-- The CXX compiler identification is GNU 4.8.5
-- Check for working CXX compiler: /usr/bin/c++
CMake Error in /root/bladebit/modules/bls-signatures/build/CMakeFiles/CMakeTmp/CMakeLists.txt:
  Target "cmTC_3cbcb" requires the language dialect "CXX17" (with compiler
  extensions), but CMake does not know the compile flags to use to enable it.

CMake Error at /usr/local/cmake/share/cmake-3.14/Modules/CMakeTestCXXCompiler.cmake:37 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  CMakeLists.txt:12 (project)

-- Configuring incomplete, errors occurred!
See also "/root/bladebit/modules/bls-signatures/build/CMakeFiles/CMakeOutput.log".

**Comment by harold-b** [comment, marked as answer]
Looks like you are using a fairly old version of gcc:
> GNU 4.8.5

Which does not support C++17 which is required by both Bladebit and BLS signatures:
https://gcc.gnu.org/onlinedocs/gcc-4.8.5/gcc/Standards.html#Standards
